internal/app/backend/backend_service.go

```diff
@@ -213,53 +223,35 @@ func matchesFromHTTPMMF(ctx context.Context, profile *pb.MatchProfile, client *h
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		errChan <- status.Error(codes.FailedPrecondition, err.Error())
-		logger.WithError(err).Error("failed to read from response body")
-		return
+		return nil, status.Errorf(codes.FailedPrecondition, "failed to read from response body for profile %s: %s", profile.Name, err.Error())
 	}
 
 	pbResp := &pb.RunResponse{}
 	err = json.Unmarshal(body, pbResp)
 	if err != nil {
-		errChan <- status.Error(codes.FailedPrecondition, err.Error())
-		logger.WithError(err).Error("failed to unmarshal response body to response pb")
-		return
+		return nil, status.Errorf(codes.FailedPrecondition, "failed to unmarshal response body to response pb for profile %s: %s", profile.Name, err.Error())
 	}
 
-	for _, match := range pbResp.Proposal {
-		matchChan <- match
-	}
+	return pbResp.Proposal, nil
 }
 
 // matchesFromGRPCMMF triggers execution of MMFs to fetch match results for each profile.
 // These proposals are then sent to evaluator and the results are streamed back on the channel
 // that this function returns to the caller.
-func matchesFromGRPCMMF(ctx context.Context, profile *pb.MatchProfile, client pb.MatchFunctionClient, matchChan chan<- *pb.Match, errChan chan<- error) {
+func matchesFromGRPCMMF(ctx context.Context, profile *pb.MatchProfile, client pb.MatchFunctionClient) ([]*pb.Match, error) {
 	// TODO: This code calls user code and could hang. We need to add a deadline here
 	// and timeout gracefully to ensure that the ListMatches completes.
-	// TODO: Currently, a failure in running the MMF is silently ignored and does not
-	// fail the FetchMatches. This needs to be revisited to investigate whether the
-	// FetchMatches should fail or atleast hint that a part of MMF executions failed.
 	resp, err := client.Run(ctx, &pb.RunRequest{Profile: profile})
 	if err != nil {
-		logger.WithFields(logrus.Fields{
-			"error":   err.Error(),
-			"profile": profile,
-		}).Error("failed to run match function for profile")
-		errChan <- err
-		return
+		logger.WithError(err).Error("failed to run match function for profile")
```

Comment:
Why does this one, unlike the others, log the error with context,  then return error without context?

Reply:
This one makes a call to the service that we own, and we should expect the error being returned contains the status code with context.